apigw_ca_handler.go

```diff
@@ -0,0 +1,110 @@
+package g8
+
+import (
+	"context"
+	"errors"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/newrelic/go-agent/_integrations/nrlambda"
+	"strings"
+
+	"github.com/JSainsburyPLC/g8/auth"
+
+	"github.com/aws/aws-lambda-go/events"
+	newrelic "github.com/newrelic/go-agent"
+	"github.com/rs/zerolog"
+)
+
+// APIGatewayCustomAuthorizerContext the context for a request for Custom Authorizer
+type APIGatewayCustomAuthorizerContext struct {
+	Context       context.Context
+	Request       events.APIGatewayCustomAuthorizerRequestTypeRequest
+	Response      auth.AuthorizerResponse
+	Logger        zerolog.Logger
+	NewRelicTx    newrelic.Transaction
+	CorrelationID string
+}
+
+// APIGatewayCustomAuthorizerHandlerFunc to populate
+type APIGatewayCustomAuthorizerHandlerFunc func(c *APIGatewayCustomAuthorizerContext) error
+
+// APIGatewayCustomAuthorizerHandler fd
+func APIGatewayCustomAuthorizerHandler(
+	h APIGatewayCustomAuthorizerHandlerFunc,
+	conf HandlerConfig,
+) func(context.Context, events.APIGatewayCustomAuthorizerRequestTypeRequest) (events.APIGatewayCustomAuthorizerResponse, error) {
+
+	return func(ctx context.Context, r events.APIGatewayCustomAuthorizerRequestTypeRequest) (events.APIGatewayCustomAuthorizerResponse, error) {
+		if len(r.MethodArn) == 0 {
+			return events.APIGatewayCustomAuthorizerResponse{}, errors.New("MethodArn is not set")
+		}
+
+		correlationID := getCorrelationIDAPIGW(r.Headers)
+
+		logger := configureLogger(conf).
+			Str("route", r.RequestContext.ResourcePath).
```

Comment:
I think the other context fields we have in the api gateway handler are still valid here - maybe we should add back the other fields here:

```
Str("application", conf.AppName).
Str("function_name", conf.FunctionName).
Str("env", conf.EnvName).
Str("build_version", conf.BuildVersion).
```

Reply:
@w32blaster hey mate - did you see this one?